GitUI/UserControls/RevisionGrid/RevisionDataGridView.cs

```diff
@@ -622,6 +623,35 @@ private void SetRowCountAndSelectRowsIfReady(int rowCount)
             SelectRowsIfReady(rowCount);
         }
 
+        private void OnScroll()
+        {
+            UpdateVisibleRowRange();
+
+            // When scrolling many rows within a short time, the message pump is
+            // flooded with WM_CTLCOLORSCROLLBAR messages and the DataGridView
+            // is not repainted. This happens for example when the mouse wheel
+            // is spinning fast (with free-spinning mouse wheels) or while dragging
+            // the scroll bar fast. In such cases, force a repaint to make the GUI
+            // feel more responsive.
+            if (_lastScroll.ElapsedMilliseconds < 100)
+            {
+                _consecutiveScroll.Start();
+            }
+            else
+            {
+                _consecutiveScroll.Reset();
+            }
+
+            if (_consecutiveScroll.ElapsedMilliseconds > 50 &&
+                _lastRepaint.ElapsedMilliseconds > 50)
```

Comment:
code style (left reading):
```suggestion
            if (_consecutiveScroll.ElapsedMilliseconds > 50
                && _lastRepaint.ElapsedMilliseconds > 50)
```